### PR TITLE
[RFR] Basic home page rendering

### PR DESCRIPTION
**Pull request by JulienMattiussi**
https://trello.com/c/xV6wZFTP

image

Simple display of the home page

- [x] Show the home page
- [x] Test button rendering

### CODE REVIEW

File: pages/globalComponents/StyledButton.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import styled from 'react-emotion';
+
+export const StyledButton = styled('button')`
+    display: inline-block;
+    -webkit-box-sizing: content-box;
```
Comment (fzaninotto) [inline pages/globalComponents/StyledButton.js]: you don't need to add browser prefixes, emotion will do it for you

File: pages/globalComponents/StyledButton.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import styled from 'react-emotion';
```
Comment (fzaninotto) [inline pages/globalComponents/StyledButton.js]: I wouldn't put it in `pages/globalComponents`, more in `/ui`, or `/chrome`

File: pages/index.js [hunk 3]
```diff
@@ -1,31 +1,22 @@
 import React from 'react';
 import Link from 'next/link';
-import styled, { css } from 'react-emotion';
+import { bodyStyle } from './styles/GlobalStyles';
```
Comment (fzaninotto) [inline pages/index.js]: Make it a styled component instead

File: pages/globalComponents/StyledButton.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+export const StyledButton = styled('button')`
```
Comment (fzaninotto) [inline pages/globalComponents/StyledButton.js]: Make it the default export instead

File: pages/index.js [hunk 5]
```diff
@@ -1,31 +1,22 @@
...
+import { StyledButton } from './globalComponents/StyledButton';
```
Comment (fzaninotto) [inline pages/index.js]: You can call it `Button`. In JavaScript, you can't have global name conflicts.

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: pages/index.js (hunk 3 above)
Comment (ThieryMichel) [reply pages/index.js]: Yes it's better to have the style with the component. It's one of the advantage of css in js

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: pages/Game.js [hunk 6]
```diff
@@ -1,20 +1,61 @@
...
+
+Game.propTypes = {
+    game: PropTypes.object.isRequired,
+    loaded: PropTypes.bool,
+};
+
+Game.getInitialProps = async function() {
+    let url = `http://${config.apiUrl}/479`;
+
+    const res = await fetch(url);
```
Comment (ThieryMichel) [inline pages/Game.js:53]: hmm, this will not work server side, use isomorphic-fetch

File: pages/index.js [hunk 7]
```diff
@@ -1,31 +1,22 @@
...
 import Link from 'next/link';
-import styled, { css } from 'react-emotion';
+import Button from '../src/ui/Button';
+import Container from '../src/ui/Container';
 
 class HomeQuarto extends React.Component {
     render() {
         return (
-            <div className={myStyle}>
+            <Container>
```
Comment (ThieryMichel) [inline pages/index.js]: use a function  component since there is no logic beside render

File: src/game/RemainingList.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+class RemainingList extends React.Component {
```
Comment (ThieryMichel) [inline src/game/RemainingList.js]: function component

File: src/ui/Button.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+const Button = styled('button')`
+    display: inline-block;
+    box-sizing: content-box;
+    cursor: pointer;
+    padding: 10px 20px;
+    border: 1px solid #018dc4;
+    border-radius: 3px;
+    font: normal medium/normal Arial, Helvetica, sans-serif;
+    color: rgba(255, 255, 255, 0.9);
+    -o-text-overflow: clip;
```
Comment (ThieryMichel) [inline src/ui/Button.js]: no need to use opera vendor, emotion should do it automaticaly

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/game/RemainingList.js (hunk 8 above)
Comment (JulienMattiussi) [reply src/game/RemainingList.js]: Actually, I already changed this in next PR

**Review by ThieryMichel — APPROVED** [review]

**Pull request merged**